snap/pack/pack.go

```diff
@@ -295,8 +295,11 @@ func mksquashfs(sourceDir, fName, snapType string, opts *Options) error {
 
 func componentPath(ci *snap.ComponentInfo, targetDir, compName string) string {
 	if compName == "" {
+		// Note that here we do not know the version of the snap, so if
+		// there is no version in component.yaml we will get names like
+		// "<snap>+<comap>_.comp"
 		// TODO should we consider architecture as with snaps?
-		compName = fmt.Sprintf("%s_%s.comp", ci.FullName(), ci.Version)
+		compName = fmt.Sprintf("%s_%s.comp", ci.FullName(), ci.Version(""))
```

Comment:
wouldn't it make more sense to leave out the _ if we have no version?

Reply:
Good point, changed now